src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Registers Twig runtime services.
+ */
+class RuntimeLoaderPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('twig.runtime_loader')) {
+            return;
+        }
+
+        $definition = $container->getDefinition('twig.runtime_loader');
+        $mapping = array();
+        foreach ($container->findTaggedServiceIds('twig.runtime') as $id => $attributes) {
+            $mapping[$container->getDefinition($id)->getClass()] = $id;
```

Comment:
you should validate that services are public (and not abstract), to provide an early error message for such mistake